types/fitty/index.d.ts

```diff
@@ -0,0 +1,41 @@
+// Type definitions for fitty 2.3
+// Project: https://github.com/rikschennink/fitty
+// Definitions by: Raiper34 <https://github.com/Raiper34>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+export as namespace fitty;
+
+declare function fitty(el: HTMLElement | string, config?: fitty.FittyConfig): fitty.FittyInstance | fitty.FittyInstance[];
```

Comment:
```suggestion
declare function fitty(selector: string, config?: fitty.FittyConfig): fitty.FittyInstance[];
declare function fitty(el: HTMLElement, config?: fitty.FittyConfig): fitty.FittyInstance;
```
literally, as per readme description. This will remove the need to use any casts or assertions

Reply:
Hey thanks you very much, and what do you think about this one?
`myFittyElement.addEventListener('fit' as keyof HTMLElementEventMap, e => console.log((e as fitty.FittyEvent).detail));`

Reply:
you would have to extend HTMLElementEventMap with your own interface that can be used in custom HTMLElement used by fitty. So your element (fifty) would have typed support for 'fit' event.
Please use Discord TypeScript channel, there are very helpful folks for those kind of problem:
https://discord.gg/SuGWHrbS

Reply:
quick & dirty:

```ts
    interface FittyEventTarget extends EventTarget {
        addEventListener(type: 'fit', listener: (this: this, ev: FittyEvent) => any, useCapture?: boolean): void;
        addEventListener(
            type: string,
            listener: EventListenerOrEventListenerObject | null,
            options?: boolean | AddEventListenerOptions,
        ): void;
        removeEventListener(type: 'fit', listener: (this: this, ev: FittyEvent) => any, useCapture?: boolean): void;
        removeEventListener(
            type: string,
            callback: EventListenerOrEventListenerObject | null,
            options?: EventListenerOptions | boolean,
        ): void;
    }
   
....
    interface FittyEvent extends Event {
        detail: FittyEventDetail;
    }

....
interface FittyInstance {
        element: HTMLElement & FittyEventTarget;
....
```